core/store/src/lib.rs

```diff
@@ -241,6 +249,51 @@ pub fn get_account(state_update: &TrieUpdate, key: &AccountId) -> Option<Account
     proto.and_then(|value| value.try_into().ok())
 }
 
+pub fn set_received_data(
+    state_update: &mut TrieUpdate,
+    account_id: &AccountId,
+    data_id: &CryptoHash,
+    data: Option<Vec<u8>>,
+) {
+    let serialized_data = match data {
+        Some(mut v) => {
+            v.push(RECEIVED_DATA_SERIALIZATION_SUFFIX);
```

Comment:
Could you remind why this is needed? Maybe even leave a comment.

Reply:
There is a comment for the constant. It's needed to distinguish between Some(vec![]) and None. Once we switch to CBOR, it will be easier.